pkg/filters/binary.go

```diff
@@ -0,0 +1,182 @@
+	if ns, err := proc.GetProcNS(1, "mnt"); err == nil {
+		return uint32(ns)
+	}
+
+	return 0
+}
+
+func NewBinaryFilter() *BinaryFilter {
+	return &BinaryFilter{
+		equal:     map[nsBinary]bool{},
+		notEqual:  map[nsBinary]bool{},
+		hostMntNS: getHostMntNS(),
+	}
+}
+
+func (f *BinaryFilter) Parse(operatorAndValues string) error {
+	if len(operatorAndValues) < 2 {
+		return InvalidExpression(operatorAndValues)
+	}
+	valuesString := string(operatorAndValues[1:])
+	operatorString := string(operatorAndValues[0])
+
+	if operatorString == "!" {
+		if len(operatorAndValues) < 3 {
+			return InvalidExpression(operatorAndValues)
+		}
+		operatorString = operatorAndValues[0:2]
+		valuesString = operatorAndValues[2:]
+	}
+
+	values := strings.Split(valuesString, ",")
+
+	for _, val := range values {
+		mntAndPath := strings.Split(val, ":")
+		var bin nsBinary
+		if len(mntAndPath) == 1 {
+			bin.path = val
+		} else if len(mntAndPath) == 2 {
+			bin.path = mntAndPath[1]
+			if mntAndPath[0] == "host" {
+				bin.mntNS = f.hostMntNS
+			} else {
+				mntNS, err := strconv.Atoi(mntAndPath[0])
+				if err != nil {
+					return InvalidValue(val)
+				}
+				bin.mntNS = uint32(mntNS)
+			}
+		} else {
+			return InvalidValue(val)
+		}
+
+		if !strings.HasPrefix(bin.path, "/") {
+			return InvalidValue(val)
+		}
+
+		err := f.add(bin, stringToOperator(operatorString))
+		if err != nil {
+			return err
+		}
+	}
+
+	f.Enable()
+
+	return nil
+}
+
+func (f *BinaryFilter) add(bin nsBinary, operator Operator) error {
+	switch operator {
+	case Equal:
+		f.equal[bin] = true
+		return nil
+	case NotEqual:
+		f.notEqual[bin] = true
+		return nil
+	default:
+		return UnsupportedOperator(operator)
+	}
+}
+
+func (f *BinaryFilter) Enable() {
+	f.enabled = true
+}
+
+func (f *BinaryFilter) Disable() {
+	f.enabled = false
+}
+
+func (f *BinaryFilter) Enabled() bool {
+	return f.enabled
+}
+
+func (f *BinaryFilter) FilterOut() bool {
+	if len(f.equal) > 0 && len(f.notEqual) == 0 {
+		return false
+	} else {
+		return true
+	}
+}
+
+type BPFBinaryFilter struct {
+	BinaryFilter
+	binaryMapName string
+}
+
+func NewBPFBinaryFilter(binaryMapName string) *BPFBinaryFilter {
+	return &BPFBinaryFilter{
+		BinaryFilter:  *NewBinaryFilter(),
+		binaryMapName: binaryMapName,
+	}
+}
+
+func (f *BPFBinaryFilter) InitBPF(bpfModule *bpf.Module) error {
+	const (
+		maxBpfBinPathSize = 256 // maximum binary path size supported by BPF
+		bpfBinFilterSize  = 264 // the value size of the BPF binary filter map entry
+	)
+
+	if !f.Enabled() {
+		return nil
+	}
+
+	binMap, err := bpfModule.GetMap(f.binaryMapName)
+	if err != nil {
+		return err
+	}
+
+	fn := func(bin nsBinary, eqVal uint32) error {
+		if len(bin.path) > maxBpfBinPathSize {
+			return InvalidValue(bin.path)
+		}
+		binBytes := make([]byte, bpfBinFilterSize)
+		if bin.mntNS == 0 {
+			copy(binBytes, bin.path)
+		} else {
+			binary.LittleEndian.PutUint32(binBytes, bin.mntNS)
+			copy(binBytes[4:], bin.path)
```

Comment:
I think you should comment here on the need to use the initial 32bit when no mountNs is given (due to map indexing). It did not occur to me naturally when reading.